crates/sui-indexer/src/framework/interface.rs

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use sui_rest_api::CheckpointData;
+use sui_types::messages_checkpoint::CheckpointSequenceNumber;
+
+#[async_trait::async_trait]
+pub trait Handler: Send {
+    fn name(&self) -> &str;
+    async fn process_checkpoint(&mut self, checkpoint_data: &CheckpointData) -> Result<()>;
+}
+
+pub trait BackfillHandler: Handler {
+    fn last_processed_checkpoint(&self) -> Option<CheckpointSequenceNumber>;
```

Comment:
if an indexer writer pod crashed and restarted, will it be a `BackfillHandler`, it also needs `last_processed_checkpoint`?

Reply:
Yeah that's likely the case, today it just queries this information out of band but it would likely be cleaner to just have last processed be something that all Handlers support. This has the side effect of making every handler one that can run in backfill mode too